### ISSUE TITLE
runtime error: invalid memory address when plan.json exists

### ISSUE THREAD

**Issue by judell** [turbot/steampipe-plugin-terraform#54]
Given this TF:

```
variable "allowed_ssh_cidrs" {
  description = "List of CIDRs that can SSH into instances"
  type        = list(string)
  default     = ["10.0.0.1/32"]
}

resource "aws_security_group" "example_sg" {
  name        = "example_sg"
  description = "Example Security Group"

  ingress {
    from_port   = 22
    to_port     = 22
    protocol    = "tcp"
    cidr_blocks = var.allowed_ssh_cidrs
  }
}
```

```
> select path from terraform_resource
+----------------------------------------------------+
| path                                               |
+----------------------------------------------------+
| /home/jon/terraform-examples/dynamic.tf |
+----------------------------------------------------+
```

Now:

```
terraform plan -var 'allowed_ssh_cidrs=["0.0.0.0/0"]' -out=plan.tfplan
terraform show -json plan.tfplan > plan.json
```

```
> select path from terraform_resource

Error: runtime error: invalid memory address or nil pointer dereference (SQLSTATE HV000)

+----------------------------------------------------+
| path                                               |
+----------------------------------------------------+
| /home/jon/terraform-examples/dynamic.tf |
+----------------------------------------------------+
```

**Comment by judell** [comment]
Actually I don't think it's var-related.

```
provider "aws" {
  region = "us-east-1"
}

resource "aws_s3_bucket" "my_bucket" {
  bucket = "jon-test-project-foo-bucket-01"
}
```

```
> select path from terraform_resource
+-----------------------------------------------------------+
| path                                                      |
+-----------------------------------------------------------+
| /home/jon/terraform-examples/bucket.tf         |
+-----------------------------------------------------------+
```
```
terraform plan -out=plan.tfplan
terraform show -json plan.tfplan > plan.json
```

```
> select path from terraform_resource

Error: runtime error: invalid memory address or nil pointer dereference (SQLSTATE HV000)
```

```
rm path.json
```

```
> select path from terraform_resource
+---------------------------------------------------+
| path                                              |
+---------------------------------------------------+
| /home/jon/terraform-examples/bucket.tf |
+---------------------------------------------------+
```

**Comment by cbruno10** [comment]
I was able to reproduce locally with both of these connection configs:
```
connection "terraform" {
 plugin = "terraform"
 plan_file_paths = ["tfplan.json"]
}

connection "terraform" {
 plugin = "terraform"
 plan_file_paths = ["*.tfplan.json"]
}
```

So it doesn't seem like the wildcard affects this particular bug (unlike https://github.com/turbot/steampipe-plugin-terraform/issues/55).

**Comment by judell** [comment]
@bigdatasourav this didn't fix #54 :-(

The above examples still fail in the same way, and nuking plan.json is still the workaround. 

My config:

```
configuration_file_paths = ["/home/jon/terraform-examples/*.tf"]
plan_file_paths          = ["/home/jon/terraform-examples/p*.json"]
state_file_paths         = ["/home/jon/terraform-examples/*.tfstate"]
```

**Comment by judell** [comment]
OK all good, I just had to use pkill -f steampipe in order to get the rebuilt branch to take effect. Thanks @bigdatasourav!
!
